src/doors/TrainDoor.cpp

```diff
@@ -0,0 +1,415 @@
+        ClassDB::bind_method(D_METHOD("get_door_platform_speed"), &TrainDoor::get_door_platform_speed);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::FLOAT, "platform/speed"), "set_door_platform_speed", "get_door_platform_speed");
+        ClassDB::bind_method(D_METHOD("set_mirror_max_shift", "mirror_max_shift"), &TrainDoor::set_mirror_max_shift);
+        ClassDB::bind_method(D_METHOD("get_mirror_max_shift"), &TrainDoor::get_mirror_max_shift);
+        ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "mirror/max_shift"), "set_mirror_max_shift", "get_mirror_max_shift");
+        ClassDB::bind_method(D_METHOD("set_mirror_vel_close", "mirror_vel_close"), &TrainDoor::set_mirror_vel_close);
+        ClassDB::bind_method(D_METHOD("get_mirror_vel_close"), &TrainDoor::get_mirror_vel_close);
+        ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "mirror/vel_close"), "set_mirror_vel_close", "get_mirror_vel_close");
+        ClassDB::bind_method(D_METHOD("set_door_needs_permit", "needs_permit"), &TrainDoor::set_door_needs_permit);
+        ClassDB::bind_method(D_METHOD("get_door_needs_permit"), &TrainDoor::get_door_needs_permit);
+        ADD_PROPERTY(PropertyInfo(Variant::BOOL, "needs_permit"), "set_door_needs_permit", "get_door_needs_permit");
+        ClassDB::bind_method(
+                D_METHOD("set_door_permit_light_blinking", "blinking_mode"),
+                &TrainDoor::set_door_permit_light_blinking);
+        ClassDB::bind_method(D_METHOD("get_door_permit_light_blinking"), &TrainDoor::get_door_permit_light_blinking);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::INT, "permit/light_blinking"), "set_door_permit_light_blinking",
+                "get_door_permit_light_blinking");
+    }
+    void TrainDoor::_do_fetch_state_from_mover(TMoverParameters *mover, Dictionary &state) {}
+
+    void TrainDoor::_do_process_mover(TMoverParameters *mover, double delta) {}
+
+    void TrainDoor::_do_update_internal_mover(TMoverParameters *mover) {
+        const std::map<std::string, control_t>::iterator open_method_lookup =
+                DoorControls.find(IntToDoorControls[door_open_method]);
+        mover->Doors.open_control =
+                open_method_lookup != DoorControls.end() ? open_method_lookup->second : control_t::passenger;
+        const std::map<std::string, control_t>::iterator close_method_lookup =
+                DoorControls.find(IntToDoorControls[door_close_method]);
+        mover->Doors.close_control =
+                close_method_lookup != DoorControls.end() ? close_method_lookup->second : control_t::passenger;
+        mover->Doors.auto_duration = door_stay_open;
+        mover->Doors.auto_velocity = door_auto_close_vel;
+        mover->Doors.auto_include_remote = door_auto_close_remote;
+        mover->Doors.permit_needed = door_needs_permit;
+        const auto permit_presets = door_permit_list.split("|");
```

Comment:
A nie lepiej zrobić TypedArray[String]? Może nie róbmy we wrapperze Szamynienia.